### PR TITLE
Add helper function to facilitate use of library with oneOf types

### PR DESCRIPTION
**Pull request by uasouz**
Any**Many thanks for submitting your Pull Request :heart:!**

**What this PR does / why we need it**:
I was using the library and it felt a by unconfortable to check every item on an oneOf if it was null or not to define it inner type so i made a minor tweak on generator and deserializers to haver a hint allowing the use of a switch expression to select the type.

Here follows a example of usage:

```java
switch (taskitem.get()) {
            case CallTask callTask -> {
                switch (callTask.get()) {
                    case CallHTTP httpCall -> {
                        yield activities.execute(HTTP_ACTION_METHOD_NAME, TaskResult.class, execution.getContext(), input);
                    }
                    case CallGRPC grpcCall -> {

                        throw new NotImplementedException();
                    }
                    case CallFunction functionCall -> {

                        throw new NotImplementedException();
                    }
                    case CallOpenAPI openApiCall -> {

                        throw new NotImplementedException();
                    }
                    case CallAsyncAPI asyncApiCall -> {

                        throw new NotImplementedException();
                    }
                    default -> throw new IllegalStateException("Unexpected value: " + callTask.get());
                }
            }
            case SwitchTask switchTask -> {
                throw new NotImplementedException();
            }
            case DoTask doTask -> {

                throw new NotImplementedException();
            }
            case ForkTask forkTask -> {

                throw new NotImplementedException();
            }
            case ForTask forTask -> {

                throw new NotImplementedException();
            }
            case RaiseTask raiseTask -> {

                throw new NotImplementedException();
            }
            case SetTask setTask -> {

                throw new NotImplementedException();
            }
            case TryTask tryTask -> {

                throw new NotImplementedException();
            }
            case WaitTask waitTask -> {

                throw new NotImplementedException();
            }
            case ListenTask listenTask -> {

                throw new NotImplementedException();
            }
            case EmitTask emitTask -> {

                throw new NotImplementedException();
            }
            default -> {
                yield null;
            }
 } 
```
**Special notes for reviewers**:

I'm open to comments and reviews and still can contribute on this PR

**Additional information (if needed):**

### CODE REVIEW

**Comment by ricardozanini** [comment]
@uasouz thank you! Can u please sign your commit?

File: custom-generator/src/main/java/io/serverlessworkflow/generator/AllAnyOneOfSchemaRule.java [hunk 1]
```diff
@@ -166,6 +166,19 @@ private JDefinedClass createUnionClass(
...
           .annotate(JsonDeserialize.class)
           .param("using", generateDeserializer(definedClass, unionTypes));
+
+      JType clazzClass = definedClass.owner()._ref(Object.class);
+
+      JFieldVar typeField =
+          definedClass.field(
+              JMod.PRIVATE,
+              clazzClass,
+              ruleFactory.getNameHelper().getPropertyName("definition", null),
```
Comment (fjtirado) [inline custom-generator/src/main/java/io/serverlessworkflow/generator/AllAnyOneOfSchemaRule.java]: I would not use `definition` property name here, just implements an interface which define a method named `unwrap`, `get` or `case`

File: api/src/main/java/io/serverlessworkflow/serialization/DeserializeHelper.java [hunk 2]
```diff
@@ -32,7 +33,12 @@ public static <T> T deserializeOneOf(
     for (Class<?> unionType : unionTypes) {
       try {
         Object object = p.getCodec().treeToValue(node, unionType);
-        return targetClass.getConstructor(unionType).newInstance(object);
+        T instance = targetClass.getConstructor(unionType).newInstance(object);
+        Field typeField = targetClass.getDeclaredField("definition");
```
Comment (fjtirado) [inline api/src/main/java/io/serverlessworkflow/serialization/DeserializeHelper.java]: Rathen than changing this class to change the private field, I would add a `this.<privateField> = xxxTask`; to every generated constructor in `AllAnyOnfOfSchemaRule`.

**Comment by uasouz** [comment]
@fjtirado Thanks for the feedback, I will adjust the PR accordingly and send new commits soon.

@ricardozanini Sorry for many force commits, had small issue with bad config not signing with mixed git configs.

**Comment by ricardozanini** [comment]
@uasouz please try squashing your commits and sign it.

**Comment by uasouz** [comment]
The PR is updated.

I squashed the commits and looks like they are signed and verified. But it still fails the check. I have to figure out why. But if there is anything else I can change on the PR, just comment.

**Comment by ricardozanini** [comment]
@uasouz since this is your first contribution, the checks only happen when I approve the run. It's a security measurement from GH.

**Review by ricardozanini — APPROVED** [review]

**Review by fjtirado — APPROVED** [review]

**Pull request merged**